app/src/main/java/com/example/androidapp/MainActivity.java

```diff
@@ -27,6 +28,7 @@
     private ActivityMainBinding binding;
     private boolean runningQOrLater =
             android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.Q;
+    public static SensorController step_sensor=null;
```

Comment:
The SensorController should be instantiated somewhere (otherwise it doesn't work, does it?).

The main idea is that the SensorController works as the main orchestrator of all the sensors (= the sensor listeners) we are going to use, so it will instantiate and manage the data they are going to generate.